src/autora/utils/transform.py

```diff
@@ -0,0 +1,150 @@
+        >>> np.array_equal(array_transformed, array_cast)
+        False
+
+        The cast array contains objects which are hard to work with:
+        >>> array_cast
+        array([[list([0, 0]), list([0, 1]), list([0, 2])],
+               [list([1, 0]), list([1, 1]), list([1, 2])],
+               [list([2, 0]), list([2, 1]), list([2, 2])]], dtype=object)
+
+        The transformed array containst vectors (numbers):
+        >>> array_transformed
+        array([[[0, 0],
+                [0, 1],
+                [0, 2]],
+        <BLANKLINE>
+               [[1, 0],
+                [1, 1],
+                [1, 2]],
+        <BLANKLINE>
+               [[2, 0],
+                [2, 1],
+                [2, 2]]])
+
+        ... the same is true for arrays:
+        >>> df_array = pd.DataFrame({
+        ...     'x_0': [np.array([0, 0]), np.array([1, 0]), np.array([2, 0])],
+        ...     'x_1': [np.array([0, 1]), np.array([1, 1]), np.array([2, 1])],
+        ...     'x_2': [np.array([0, 2]), np.array([1, 2]), np.array([2, 2])]
+        ... })
+        >>> array_transformed = to_array(df_array)
+        >>> array_cast = np.array(df_list)
+
+        the results are not equal:
+        >>> np.array_equal(array_transformed, array_cast)
+        False
+
+        The cast array contains objects which are hard to work with:
+        >>> array_cast
+        array([[list([0, 0]), list([0, 1]), list([0, 2])],
+               [list([1, 0]), list([1, 1]), list([1, 2])],
+               [list([2, 0]), list([2, 1]), list([2, 2])]], dtype=object)
+
+        The transformed array containst vectors (numbers):
+        >>> array_transformed
+        array([[[0, 0],
+                [0, 1],
+                [0, 2]],
+        <BLANKLINE>
+               [[1, 0],
+                [1, 1],
+                [1, 2]],
+        <BLANKLINE>
+               [[2, 0],
+                [2, 1],
+                [2, 2]]])
+
+        # This also works with more nesting:
+        >>> df_nested = pd.DataFrame({
+        ...     'x_0': [[[0,0],[1,1]], [[0,0],[2,2]]],
+        ...     'x_1': [[[1,1],[1,1]], [[1,1],[2,2]]]
+        ... })
+        >>> to_array(df_nested)
+        array([[[[0, 0],
+                 [1, 1]],
+        <BLANKLINE>
+                [[1, 1],
+                 [1, 1]]],
+        <BLANKLINE>
+        <BLANKLINE>
+               [[[0, 0],
+                 [2, 2]],
+        <BLANKLINE>
+                [[1, 1],
+                 [2, 2]]]])
+
+        When the inner lists don't have the same shape, an error is thrown and one can use
+        a flattening version of this (ATTENTION: when using the flattening version,
+        information about which entry belongs to which condition is lost):
+    """
+    if isinstance(arr, np.ndarray):
+        return arr
+
+    _lst = [list(row) for _, row in arr.iterrows()]
+    return np.array(_lst)
+
+
+def to_array_flatten(arr: Union[pd.DataFrame, pd.Series, np.ndarray]) -> np.ndarray:
```

Comment:
Here, we could either consider having a separate function `flatten` which can be combined with the `to_array` function. Alternatively, we could call it `to_flat_array`?